source/patterns/@aws-solutions-constructs/aws-apigatewayv2websocket-sqs/lib/index.ts

```diff
@@ -0,0 +1,178 @@
+
+  /**
+   * Optional user-provided props to override the default props for the log group.
+   *
+   * @default - Default props are used
+   */
+  readonly logGroupProps?: logs.LogGroupProps;
+  /**
+   * If no key is provided, this flag determines whether the queue is encrypted with a new CMK or an AWS managed key.
+   * This flag is ignored if any of the following are defined: queueProps.encryptionMasterKey, encryptionKey or encryptionKeyProps.
+   *
+   * @default - False if queueProps.encryptionMasterKey, encryptionKey, and encryptionKeyProps are all undefined.
+   */
+  readonly enableEncryptionWithCustomerManagedKey?: boolean;
+  /**
+   * An optional, imported encryption key to encrypt the SQS Queue with.
+   *
+   * @default - None
+   */
+  readonly encryptionKey?: kms.Key;
+  /**
+   * Optional user provided properties to override the default properties for the KMS encryption key used to encrypt the SQS queue with.
+   *
+   * @default - None
+   */
+  readonly encryptionKeyProps?: kms.KeyProps;
+  /**
+   * Whether to deploy a secondary queue to be used as a dead letter queue.
+   *
+   * @default - required field.
+   */
+  readonly deployDeadLetterQueue?: boolean;
+  /**
+   * Optional user provided properties for the dead letter queue
+   *
+   * @default - Default props are used
+   */
+  readonly deadLetterQueueProps?: sqs.QueueProps;
+  /**
+   * The number of times a message can be unsuccessfully dequeued before being moved to the dead-letter queue.
+   *
+   * @default - required only if deployDeadLetterQueue = true.
+   */
+  readonly maxReceiveCount?: number;
+  /**
+   * API Gateway Request Template for the default route.
+   *
+   * @default - "Action=ReceiveMessage"
+   */
+  readonly defaultRouteRequestTemplate?: { [contentType: string]: string };
+
+  /**
+   * Whether to create a default route. If set to true, then it will use the value supplied with `defaultRouteRequestTemplate`
+   *
+   * @default - false.
+   */
+  readonly createDefaultRoute?: boolean;
+
+  /**
+   * User provided websocket authorizer. If not provided, the constructs defaults to
+   * {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigatewayv2.WebSocketAuthorizer.html WebSocketAuthorizer}
+   */
+  readonly defaultAuthorizer?: apigwv2.IWebSocketAuthorizer;
+}
+
+export class ApiGatewayV2WebSocketToSqs extends Construct {
+  public readonly webSocketApi: apigwv2.WebSocketApi;
+  public readonly webSocketStage: apigwv2.WebSocketStage;
+  public readonly apiGatewayRole: iam.Role;
+  public readonly apiGatewayLogGroup: logs.LogGroup;
+  public readonly sqsQueue: sqs.Queue;
+  public readonly deadLetterQueue?: sqs.DeadLetterQueue;
+
+  constructor(scope: Construct, id: string, props: ApiGatewayV2WebSocketToSqsProps) {
+    super(scope, id);
+
+    if (props.existingWebSocketApi && props.webSocketApiProps) {
+      throw new Error("Provide either an existing WebSocketApi instance or WebSocketApiProps, not both");
+    }
```

Comment:
I quickly glanced at the unit tests, but what about the case when neither `existingWebSocketApi` or `webSocketApiProps` is supplied? Also, this is more a question for Biff, but should we put these types of validations into a defaults.CheckProps instead?

Reply:
Unit test added.